File: c/functions/recursion.c
#include <stdio.h>

int factorial(int);

int factorial(int n)
{
    if (n == 1 || n == 0)
    { // base condition
        return 1;
    }
    return factorial(n - 1) * n;
}
int main()
{
    int a;
    printf("Enter the value of number you wanna know factorial of :");
    scanf("%d", &a);
    factorial(a);
    printf("The factorial of %d is %d \n", a, factorial(a));
    return 0;
}

File: python/functions/recursion.py
def factorial(n) :

    if(n==0 or n==1):     #Base condition is very important.... it will stop calling function further anymore..
        return 1
    else:
     return n*factorial(n-1)
    
x=int(input("Enter value of n to find value of n! : "))
print(f"n!={factorial(x)}")